contk/dataloader/dataloader.py

```diff
@@ -26,3 +30,255 @@ def load_class(cls, class_name):
+
+		Arguments:
+				key (str): must be contained in `key_name`
+				ignore_left_samples (bool): Ignore the last batch, whose sample num
+						is not equal to `batch_size`. Default: `False`
+
+		Returns:
+				A dict like :func:`get_batch`, or None if the epoch is end.
+		'''
+		if key not in self.key_name:
+			raise ValueError("No set named %s." % key)
+		if self.batch_size[key] is None:
+			raise RuntimeError( \
+				"Please run restart before calling this function.")
+		batch_id = self.batch_id[key]
+		start, end = batch_id * \
+			self.batch_size[key], (batch_id + 1) * self.batch_size[key]
+		if start >= len(self.index[key]):
+			return None
+		if ignore_left_samples and end > len(self.index[key]):
+			return None
+		index = self.index[key][start:end]
+		res = self.get_batch(key, index)
+		self.batch_id[key] += 1
+		return res
+
+	def sen_to_index(self, sen):
+		'''Convert a sentences from string to index representation.
+
+		Arguments:
+			sen (list): a list of str, representing each token of the sentences.
+
+		Examples:
+			vocab_list = ["<pad>", "<unk>", "<go>", "<eos>", "I", "have",\
+							"been", "to", "Sichuan"]
+			>>> dataloader.sen_to_index(
+			...		["<go>", "I", "have", "been", "to", "Sichuan", "<eos>"])
+			>>> [2, 4, 5, 6, 7 ,8 ,3]
+
+		'''
+		return list(map(lambda word: self.word2id.get(word, self.unk_id), sen))
+
+	def trim_index(self, index):
+		'''Trim index. There will be two steps:
+			* If there is an `<eos>` in sentences, \
+				find first `<eos>` and abondon words after it (included the `<eos>`).
+			* ignore `<pad>` s at the end of the sentence.
+
+		Arguments:
+			index (list): a list of int
+
+		Examples:
+			vocab_list = ["<pad>", "<unk>", "<go>", "<eos>", "I", "have",\
+							"been", "to", "Sichuan"]
+			>>> dataloader.trim_index(
+			...		[2, 4, 5, 6, 7, 8, 0, 0, 3, 4, 3, 0]) # <go> I have been to Sichuan <pad> <pad> <eos> I <eos> <pad>
+			>>> [2, 4, 5, 6, 7, 8] # <go> I have been to Sichuan
+		'''
+
+		index = trim_before_target(list(index), self.eos_id if not self.multi_turn else self.eot_id)
+		idx = len(index)
+		while idx > 0 and index[idx-1] == self.pad_id:
+			idx -= 1
+		index = index[:idx]
+		return index
+
+	def index_to_sen(self, index, trim=True):
+		'''Convert a sentences from index to string representation
+
+		Arguments:
+				index (list): a list of int
+				trim (bool): if True, call :func:`trim_index` before convertion.
+
+		Examples:
+			vocab_list = ["<pad>", "<unk>", "<go>", "<eos>", "I", "have",\
+							"been", "to", "Sichuan"]
+			>>> dataloader.index_to_sen(
+			...		[2, 4, 5, 6, 7, 8, 3, 0, 0], trim = True)
+			>>> ["<go>", "I", "have", "been", "to", "Sichuan"]
+			>>> dataloader.index_to_sen(
+			...		[2, 4, 5, 6, 7, 8, 3, 0, 0], trim = False)
+			>>> ["<go>", "I", "have", "been", "to", "Sichuan", "<eos>", "<pad>", "<pad>"]
+
+		'''
+		if trim:
+			index = self.trim_index(index)
+		return list(map(lambda word: self.vocab_list[word], index))
+
+	def multi_turn_trim_index(self, index):
```

Comment:
`multi_turn` is not a common function ...

Reply:
Move it to ``MultiTurnDialog`` for now, this function may be confusing for ``SingleDialog``. If we need this, we can move it back later.

Reply:
OK